src/app/clusters/bindings/bindings.cpp

```diff
@@ -25,133 +25,63 @@
 #include <app/CommandHandler.h>
 #include <app/ConcreteCommandPath.h>
+#include <app/binding/BindingTable.h>
 #include <app/util/binding-table.h>
 #include <lib/support/logging/CHIPLogging.h>
 
 using namespace chip;
 using namespace chip::app::Clusters::Binding;
+using chip::app::BindingTable;
 
-EmberStatus prepareBinding(EmberBindingTableEntry & binding, NodeId nodeId, GroupId groupId, EndpointId endpointId,
-                           ClusterId clusterId)
-{
-    if (groupId && nodeId)
-    {
-        return EMBER_BAD_ARGUMENT;
-    }
-
-    binding.clusterId    = clusterId;
-    binding.local        = emberAfCurrentCommand()->apsFrame->destinationEndpoint;
-    binding.networkIndex = 0;
+namespace {
+// TODO: Add peer fabric index to command path
+constexpr FabricIndex kPeerFabricIndex = 1;
 
-    if (groupId)
+EmberAfStatus MapBindingErrorToStatus(CHIP_ERROR error)
+{
+    if (error == CHIP_ERROR_NO_MEMORY)
     {
-        binding.type    = EMBER_MULTICAST_BINDING;
-        binding.groupId = groupId;
-        binding.remote  = 0;
+        return EMBER_ZCL_STATUS_LIMIT_REACHED;
     }
-    else
+    if (error == CHIP_NO_ERROR)
     {
-        binding.type   = EMBER_UNICAST_BINDING;
-        binding.nodeId = nodeId;
-        binding.remote = endpointId;
+        return EMBER_ZCL_STATUS_SUCCESS;
     }
-
-    return EMBER_SUCCESS;
-}
-
-EmberStatus getBindingIndex(EmberBindingTableEntry & newEntry, uint8_t * bindingIndex)
-{
-    EmberBindingTableEntry currentEntry;
-    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    if (error == CHIP_ERROR_KEY_NOT_FOUND)
     {
-        emberGetBinding(i, &currentEntry);
-        if (currentEntry.type != EMBER_UNUSED_BINDING && currentEntry == newEntry)
-        {
-            *bindingIndex = i;
-            return EMBER_SUCCESS;
-        }
+        return EMBER_ZCL_STATUS_NOT_FOUND;
     }
-
-    return EMBER_NOT_FOUND;
+    return EMBER_ZCL_STATUS_FAILURE;
 }
 
-EmberStatus getUnusedBindingIndex(uint8_t * bindingIndex)
-{
-    EmberBindingTableEntry currentEntry;
-    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
-    {
-        emberGetBinding(i, &currentEntry);
-        if (currentEntry.type == EMBER_UNUSED_BINDING)
-        {
-            *bindingIndex = i;
-            return EMBER_SUCCESS;
-        }
-    }
-
-    return EMBER_NOT_FOUND;
-}
+} // namespace
 
 bool emberAfBindingClusterBindCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
                                        const Commands::Bind::DecodableType & commandData)
 {
-    auto & nodeId     = commandData.nodeId;
-    auto & groupId    = commandData.groupId;
-    auto & endpointId = commandData.endpointId;
-    auto & clusterId  = commandData.clusterId;
-
-    ChipLogDetail(Zcl, "RX: BindCallback");
-
-    EmberBindingTableEntry bindingEntry;
-    if (prepareBinding(bindingEntry, nodeId, groupId, endpointId, clusterId) != EMBER_SUCCESS)
```

Comment:
So here is a question.  why are we rewriting the existing binding table instead of just adding fabric index to it?  It's not a huge deal, but the new implementation is broken for various cases (like groups) that the existing table handles....